equinix/resource_metal_device.go

```diff
@@ -410,6 +410,11 @@ func resourceMetalDevice() *schema.Resource {
 					},
 				},
 			},
+			"sos_hostname": {
```

Comment:
nice! we'll want to include this in the datasource and the docs/ for both resources.

Reply:
The datasource gains this attribute in #291; I broke these apart to keep smaller changes, but that was because I was worried about fixing the mock tests as part of the conversion.  Now that #371 is in, this could probably be collapsed back into #291

Reply:
Oh right, the other reason this PR is small is because so far it only converts the read function.  If we're good with using 2 clients in a single resource, though, maybe it's worth combining this with the device data source PR and then follow up in a separate PR to convert the create/update/delete functions.